test/functional/wallet_assumeutxo.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test for assumeutxo wallet related behavior.
+See feature_assumeutxo.py for background.
+
+## Possible test improvements
+
+- TODO: the "assumed" field for a confirmed wallet transaction should be
+        present during background sync, start false and become true as
+        sync progresses.
+        Alternatively prevent loading a wallet with a birthdate before
+        the snapshot height, as is done with restoring from backup.
+
+"""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+START_HEIGHT = 199
+SNAPSHOT_BASE_HEIGHT = 299
+FINAL_HEIGHT = 399
+
+class AssumeutxoTest(BitcoinTestFramework):
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def add_options(self, parser):
+        self.add_wallet_options(parser, legacy=False)
+
+    def set_test_params(self):
+        """Use the pregenerated, deterministic chain up to height 199."""
+        self.num_nodes = 2
+        self.rpc_timeout = 120
+        self.extra_args = [
+            [],
+            ["-fastprune", "-prune=1"]
+        ]
+
+    def setup_network(self):
+        """Start with the nodes disconnected so that one can generate a snapshot
+        including blocks the other hasn't yet seen."""
+        self.add_nodes(2)
+        self.start_nodes(extra_args=self.extra_args)
+
+    def run_test(self):
+        """
+        Bring up two (disconnected) nodes, mine some new blocks on the first,
+        and generate a UTXO snapshot.
+
+        Load the snapshot into the second, ensure it syncs to tip and completes
+        background validation when connected to the first.
+        """
+        n0 = self.nodes[0]
+        n1 = self.nodes[1]
+
+        # Mock time for a deterministic chain
+        for n in self.nodes:
+            n.setmocktime(n.getblockheader(n.getbestblockhash())['time'])
+
+        self.sync_blocks()
+
+        n0.createwallet('test',blank=True, disable_private_keys=True, descriptors=True)
+        w = n0.get_wallet_rpc("test")
+
+        w.importdescriptors([{"desc":"addr(mjTkW3DjgyZck4KbiRusZsqTgaYTxdSz6z)#jdtmxdcm","timestamp":0,"label":"det_coinbase_key_0"}])
+
+        # Generate a series of blocks that `n0` will have in the snapshot,
+        # but that n1 doesn't yet see. In order for the snapshot to activate,
+        # though, we have to ferry over the new headers to n1 so that it
+        # isn't waiting forever to see the header of the snapshot's base block
+        # while disconnected from n0.
+        for i in range(100):
```

Comment:
In 0b8143396183d23e215fc90a2d24813ad1b7ad23: nit
```suggestion
        for _ in range(100):
```